### PR TITLE
feat(spanner): add cmek db support - ext of PR#8139

### PR DESCRIPTION
**Pull request by jiren**
This is same code base of https://github.com/googleapis/google-cloud-ruby/pull/8139 created by @hengfengli  with added doctest and examples.

### CODE REVIEW

**Comment by jiren** [comment]
@skuruppu Created PR https://github.com/googleapis/google-cloud-ruby/pull/8167 with doctest updates and example.

**Review by skuruppu — APPROVED** [review]
Thanks a lot for putting this together at short notice @jiren.

**Comment by skuruppu** [comment]
@jiren you can rebase this PR now since the code generation PR has been merged.

**Comment by jiren** [comment]
> @jiren you can rebase this PR now since the code generation PR has been merged.

Done

File: google-cloud-spanner/lib/google/cloud/spanner/database.rb [hunk 1]
```diff
@@ -113,6 +113,14 @@ def state
...
         end
 
+        # An encryption configuration describing the encryption type and key
+        # resources in Cloud KMS.
+        #
+        # @return [Google::Cloud::Spanner::Admin::Database::V1::EncryptionConfig, nil]
+        def encryption_config
+          @grpc.encryption_config
+        end
+
```
Comment (larkee) [inline google-cloud-spanner/lib/google/cloud/spanner/database.rb:123]: The CMEK change also included adding an `encryption_info` field to the `Database` message. See [here](https://github.com/googleapis/googleapis/commit/f829b1334cce86aa3738f3c0698d814b56664445#diff-9422939aee7fc2d7c95d6282558c7480d88ce72964608f9bf7c1c79dfe1fbb54R392-R403). Note that it is a repeated field and so will need to be treated as an array.

**Review by larkee — COMMENTED** [review]
Just noticed some minor missing functionality

File: google-cloud-spanner/lib/google/cloud/spanner/database.rb (hunk 1 above)
Comment (jiren) [reply google-cloud-spanner/lib/google/cloud/spanner/database.rb:123]: Got it.

**Review by thiagotnunes — APPROVED** [review]

**Review by larkee — APPROVED** [review]

**Comment by jiren** [comment]
@skuruppu @thiagotnunes @larkee Please do not merge. Adding pending integration tests cases.

**Comment by jiren** [comment]
@larkee  updated test.

Mac tests are failing due to know issue specific to ruby 3.0.0
https://github.com/grpc/grpc/issues/25222

**Pull request merged**